src/interfaces/controllers/deregister-trailer/deregister-trailer.ts

```diff
@@ -0,0 +1,54 @@
+import { Request, Response, NextFunction } from 'express';
+import * as domain from '../../../domain';
+import { DataAccess } from '../../../utils/data-access';
+import { IRequestHandler } from '../i-request-handler';
+import { deregisterTrailerValidator } from './deregister-trailer-validator';
+
+export class DeregisterTrailer implements IRequestHandler {
+  private validate(payload?: domain.DeregisterTrailer): string {
+    const errors = deregisterTrailerValidator.validate(payload, { abortEarly: false }).error;
+    return errors?.message;
+  }
+
+  public async call(req: Request, res: Response, next: NextFunction): Promise<void> {
+    const deregisterTrailer = req.body as domain.DeregisterTrailer;
+    const { trn } = req.params;
+
+    const errors = this.validate(deregisterTrailer);
+    if (errors) {
+      next(new domain.HTTPError(400, errors));
+      return;
+    }
+
+    try {
+      const { reasonForDeregistration, deregisterDate } = deregisterTrailer;
+      let existingTrailerRegistration = await this.checkForExistingTrailerRegistration(trn);
+      if (existingTrailerRegistration) {
+        existingTrailerRegistration = {
+          ...existingTrailerRegistration,
+          reasonForDeregistration,
+          deregisterDate,
+        };
+        await DataAccess.getInstance().upsertTrailerRegistration(existingTrailerRegistration);
```

Comment:
If we have been asked to deregister a trailer, and we don't have a record for it - something has probably gone wrong somewhere upstream (maybe the register failed for some reason) Should we log an Info to say that we are ignoring a request to deregister a trailer that we don't know about?

Reply:
a warning log added and short circuit logic added

Reply:
A warning log is added.